frontend/pages/queries/QueryPage/QueryPage.jsx

```diff
@@ -83,9 +84,9 @@ export class QueryPage extends Component {
   }
 
   componentWillMount () {
-    const { dispatch, hostIDs, selectedHosts, selectedTargets } = this.props;
+    const { dispatch, hostIDs, hostUUIDs, selectedHosts, selectedTargets } = this.props;
 
-    if (hostIDs) {
+    if (hostIDs || hostUUIDs) {
```

Comment:
I think this should work. `.length` is a proto on both strings and arrays - this conditional should check for any length on either variable. Otherwise, something like an erroneous empty array would trigger a truthy result in the current conditional.

```suggestion
    if ((hostIDs.length || hostUUIDs.length) > 0) {
```

Reply:
I used a modified version of this because these variables are sometimes undefined (when no argument is specified at all) and would throw exceptions.